src/Rules/Symfony/ContainerInterfacePrivateServiceRule.php

```diff
@@ -43,11 +44,17 @@ public function processNode(Node $node, Scope $scope): array
 
 		$argType = $scope->getType($node->var);
 
-		$isTestContainerType = (new ObjectType('Symfony\Bundle\FrameworkBundle\Test\TestContainer'))->isSuperTypeOf($argType);
+		if (
+			$argType instanceof TestContainerType
```

Comment:
Could we simplify this and not report an error when we simply have the upper-most ContainerInterface and are in a subclass of KernelTestCase?

The custom type is wrong since it doesn't implement equals, describe and isSuperTypeOf. But I think we don't need it at all.

Reply:
> Could we simplify this and not report an error when we simply have the upper-most ContainerInterface and are in a subclass of KernelTestCase?

I think this is not fully true, but this might be good enough.

Reply:
Should be better with the last commit.